bundles/framework/featuredata/plugin/FeaturedataPlugin.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Message } from 'oskari-ui';
+import { ThemeProvider } from 'oskari-ui/util';
+import { FeatureDataButton } from './FeatureDataButton';
+import { FeatureDataPluginHandler } from './FeatureDataPluginHandler';
+
+/**
+ * @class Oskari.mapframework.bundle.featuredata.plugin.FeaturedataPlugin
+ * Provides WFS grid link on top of map
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.featuredata.plugin.FeaturedataPlugin',
+    /**
+     * @method create called automatically on construction
+     * @static
+     * @param {Object} config
+     *      JSON config with params needed to run the plugin
+     */
+    function () {
+        var me = this;
+        me._clazz = 'Oskari.mapframework.bundle.featuredata.plugin.FeaturedataPlugin';
+        me._defaultLocation = 'top right';
+        me._index = 100;
+        me._name = 'FeaturedataPlugin';
+        me._mapStatusChanged = true;
+        me._flyoutOpen = undefined;
+    }, {
+        _startPluginImpl: function () {
+            this.setElement(this._createControlElement());
+            this.addToPluginContainer(this.getElement());
+            this.refresh();
+        },
+        /**
+         * @method _createControlElement
+         * @private
+         * Creates container for UI for feature data plugin
+         */
+        _createControlElement: function () {
+            return jQuery('<div class="mapplugin featuredataplugin"></div>');
+        },
+        /**
+         * Handle plugin UI and change it when desktop / mobile mode
+         * @method  @public redrawUI
+         * @param  {Boolean} mapInMobileMode is map in mobile mode
+         * @param {Boolean} forced application has started and ui should be rendered with assets that are available
+         */
+        redrawUI: function (mapInMobileMode, forced) {
+            this.refresh();
+        },
+        /**
+         * @method  @public mapStatusChanged map status changed
+         * @param  {Boolean} changed is map status changed
+         */
+        mapStatusChanged: function (changed = true) {
+            this._mapStatusChanged = changed;
+        },
+
+        getMapStatusChanged: function () {
+            return this._mapStatusChanged;
+        },
+        /**
+         * @method refresh
+         * Updates the plugins interface (hides if no featuredata layer selected)
+         */
+        refresh: function () {
+            if (!this.handler) {
+                this.handler = new FeatureDataPluginHandler(this, this.getMapModule(), this.getConfig());
```

Comment:
Nothing changes `this._flyoutOpen` here so the button never has "active" color. We could add a state listener here for it to make it easier to read when the renderButton() is called:
```
this.handler.addStateListener(state => {
  // toggle  _flyoutOpen and/or loading
  // based on state and call renderButton() 
})
```

Reply:
Removed the old (copy&paste)stuff concerning flyoutOpen and set a flag in state instead.